### DISCUSSION TITLE
Shuttle engine/ballast physics of a custom sub are non-responsive in multiplayer

### DISCUSSION THREAD

**Discussion by Tsunder** [FakeFishGames/Barotrauma#14966] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

played pvp on a player hosted server w/ a shuttle attached to submarine
shuttle can undock, but engine and ballast were non-responsive.
the pumps and engine displayed they were getting proper signal (ie., the thrust/speeds were changing) but 
1. the shuttle didn't move respectively to the thrust
2. the ballast would constnatly be pumping to its target level, then suddenly be nearly full/nearly empty again and lose all of its "progress"

### Reproduction steps

1. pvp sub vs sub mode
2. both teams select submarine w/ shuttles attached. we used the Rally, which has a locally piloted shuttle, but a vanilla sub's remote drone might work too.
3. shuttle was uncontrollable

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.17.0 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! The vanilla subs don't seem to have this issue, but Rally does seem to be affected. To me it seems like it has something to do with the "manual control" button on the shuttle nav terminal (it seems to send out a signal of 0, which doesn't correctly toggle the relay?), but unfortunately it's hard for me to diagnose where exactly the issue is since the shuttle isn't available as a standalone file I could open in the sub editor.

**Comment by Tsunder** [comment]
Yalson shuttle attached

Some checking myself appears that the relay responsible for prodiving power to the junction boxes is not truly state:on at the round start, despite the UI saying it is.

~~clicking "is on" for the culprit relay to turn it "off" seems to update the script, as the NOT should turn it back on and then it provides power again.~~

I have attachehd the shuttle as "yalson debug" , with the relevant relays (the right-most one in the Relay-Not-Relay) moved outside of the wall for viewing.

there is another relay inside the sub that needed a manual pressing of "is on" (turning it off only for the NOT to turn it back on) to send power back through to another part of the shuttle.

[Yalson.zip](https://github.com/user-attachments/files/17593420/Yalson.zip)

**Comment by Tsunder** [comment]
When the relays are set to "ON" in editor the relays are on. If the relays are set to OFF in editor and thusly are supposed be turned on by the NOT components in the first update()s, the relays fail to turn on.

**Comment by Tsunder** [comment]
I think this shuttle's relays were originally placed back when relays were off by default, IIRC

**Reply by Tsunder** [reply]
simply turning all of the relays on in editor didn't fully solve this power issue though. the JBs then only display receiving half power. Unwiring the NOTs from the relays does, but it's unclear to me the causes of this...

**Reply by Tsunder** [reply]
that's weird.
Turning just the after-NOT relays on resutls in full voltage, but if all relays are set as ON in editor then the sub has half voltage.
The relays that are supposed to be OFF from the docking port being closed aren't off in some parts of code, it seems.

**Comment by Regalis11** [comment]
Seems to be caused by https://github.com/FakeFishGames/Barotrauma/discussions/15030, closing this one as a duplicate since the newer ticket has a bit more information.
